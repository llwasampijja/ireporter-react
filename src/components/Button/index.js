import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const {
    buttonClass,
    buttonId,
    buttonName,
    buttonEvent,
  } = props;
  return (
    <button
      className={buttonClass}
      id={buttonId}
      onClick={buttonEvent}
      type="button"
    >
      {buttonName}
    </button>
  );
};

Button.propTypes = {
  buttonClass: PropTypes.string,
  buttonId: PropTypes.string,
  buttonName: PropTypes.string,
  buttonEvent: PropTypes.func,
};

Button.defaultProps = {
  buttonClass: '',
  buttonId: '',
  buttonName: 'button',
  buttonEvent: () => { },
};


export default Button;
