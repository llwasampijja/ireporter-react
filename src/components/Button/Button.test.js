import React from 'react';
import { shallow } from 'enzyme';
import Button from '.';

describe('tests for the button component', () => {
  it('button renders properly', () => {
    const component = <Button />;
    expect(component).toMatchSnapshot();
  });
});
