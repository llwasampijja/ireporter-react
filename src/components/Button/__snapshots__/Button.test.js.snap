// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for the button component button renders properly 1`] = `
<Button
  buttonClass=""
  buttonEvent={[Function]}
  buttonId=""
  buttonName="button"
/>
`;
